libs/wingc/src/type_check.rs

```diff
@@ -1631,82 +1636,111 @@ impl<'a> TypeChecker<'a> {
 	/// # Returns
 	/// The hydrated type reference
 	///
-	fn hydrate_class_type_arguments(&mut self, env: &SymbolEnv, original_fqn: &str, type_param: TypeRef) -> TypeRef {
+	fn hydrate_class_type_arguments(
+		&mut self,
+		env: &SymbolEnv,
+		original_fqn: &str,
+		type_params: Vec<TypeRef>,
+	) -> TypeRef {
 		let original_type = env
 			.lookup_nested_str(original_fqn, true, None)
 			.unwrap()
 			.as_type()
 			.unwrap();
 		let original_type_class = original_type.as_class().unwrap();
+		let original_type_params = if let Some(tp) = original_type_class.type_parameters.as_ref() {
+			tp
+		} else {
+			return self.general_type_error(format!("\"{}\" does not have type parameters", original_fqn));
```

Comment:
Maybe this only needs to be an only error if `type_params.length > 0` and I would also make the error a bit more descriptive.

(also, kind of feels like this is some kind of a compiler internal error and not an error the developer can do something about)

Reply:
True, this should actually be a panic. Will update